sdk/identity/azure-identity/src/main/java/com/azure/identity/DeviceCodeCredential.java

```diff
@@ -58,7 +57,7 @@
                 return Mono.empty();
             }
         }).switchIfEmpty(
-            Mono.defer(() -> identityClient.authenticateWithDeviceCode(request, deviceCodeChallengeConsumer)))
+            Mono.defer(() -> identityClient.authenticateWithDeviceCode(request, challengeConsumer)))
```

Comment:
It's possible to have a null challengeConsumer. The calling code may throw a nullpointerexception.

Reply:
It's not possible through the validation in the builder:
```java
    public DeviceCodeCredential build() {
        ValidationUtil.validate(getClass().getSimpleName(), new HashMap<String, Object>() {{
                put("clientId", clientId);
                put("challengeConsumer", challengeConsumer);
            }});
        return new DeviceCodeCredential(clientId, tenantId, challengeConsumer, identityClientOptions);
    }
```

Reply:
Ahh! I see the ValidationUtil now. Thanks. :)